public/index.js

```diff
@@ -45,7 +45,7 @@ socket.on('disconnect', function () {
     loaded = false;
 });
 socket.on("updatenew", (data) => {
-    if (data.status == "private") {
+    if (data.status == "private" || data.status == "restricted") {
         console.log("NEW ONE HAS GONE, SO LONG");
         dark++;
```

Comment:
something i’ve just noticed — there needs to be a check here to only increment the `dark` *if the subreddit isn’t already dark* (i.e., if it’s transitioned from private to restricted/from restricted to private).

Reply:
Isnt that done backend when before it emits the socket? 

[here](https://github.com/YoloFTW/reddark/blob/824c79e2260134b4669d23cf9d2fb93f70114a48/main.js#LL159C24-L159C155) and [here](https://github.com/YoloFTW/reddark/blob/824c79e2260134b4669d23cf9d2fb93f70114a48/main.js#LL173C25-L173C196) the if statements check to see if the status isnt already the one its checking for.

specifically `subreddits[section][subreddit].status != "restricted"`.

I could be wrong and happy im happy to discuss it further